pkg/backup/item_collector.go

```diff
@@ -291,53 +291,88 @@ func (r *itemCollector) getResourceItems(log logrus.FieldLogger, gv schema.Group
-			// Use Velero's dynamic list function instead of the default
-			listPager := pager.New(pager.SimplePageFunc(func(opts metav1.ListOptions) (runtime.Object, error) {
-				return resourceClient.List(opts)
-			}))
-			// Use the page size defined in the server config
-			// TODO allow configuration of page buffer size
-			listPager.PageSize = int64(r.pageSize)
-			// Add each item to temporary slice
-			list, paginated, err := listPager.List(context.Background(), metav1.ListOptions{LabelSelector: labelSelector})
-			if err != nil {
-				log.WithError(errors.WithStack(err)).Error("Error listing resources")
-				continue
-			}
-			if !paginated {
-				log.Infof("list for groupResource %s was not paginated", gr)
-			}
-			err = meta.EachListItem(list, func(object runtime.Object) error {
-				u, ok := object.(*unstructured.Unstructured)
-				if !ok {
-					log.WithError(errors.WithStack(fmt.Errorf("expected *unstructured.Unstructured but got %T", u))).Error("unable to understand entry in the list")
-					return fmt.Errorf("expected *unstructured.Unstructured but got %T", u)
+			for _, label := range orLabelSelectors {
+				// process pager client calls
+				list, err := r.processPagerClientCalls(gr, label, resourceClient)
+				if err != nil {
+					continue
+				}
+
+				err = meta.EachListItem(list, func(object runtime.Object) error {
+					u, ok := object.(*unstructured.Unstructured)
+					if !ok {
+						log.WithError(errors.WithStack(fmt.Errorf("expected *unstructured.Unstructured but got %T", u))).Error("unable to understand entry in the list")
+						return fmt.Errorf("expected *unstructured.Unstructured but got %T", u)
+					}
+					unstructuredItems = append(unstructuredItems, *u)
+					return nil
+				})
+				if err != nil {
+					log.WithError(errors.WithStack(err)).Error("unable to understand paginated list")
+					continue
 				}
-				unstructuredItems = append(unstructuredItems, *u)
-				return nil
-			})
-			if err != nil {
-				log.WithError(errors.WithStack(err)).Error("unable to understand paginated list")
-				continue
 			}
 		} else {
-			// If limit is not positive, do not use paging. Instead, request all items at once
-			unstructuredList, err := resourceClient.List(metav1.ListOptions{LabelSelector: labelSelector})
-			if err != nil {
-				log.WithError(errors.WithStack(err)).Error("Error listing items")
-				continue
+			for _, label := range orLabelSelectors {
+				unstructuredList, err := resourceClient.List(metav1.ListOptions{LabelSelector: label})
+				if err != nil {
+					log.WithError(errors.WithStack(err)).Error("Error listing items")
+					continue
+				}
+				unstructuredItems = append(unstructuredItems, unstructuredList.Items...)
+			}
+
+		}
+
+		var labelSelector string
+		if selector := r.backupRequest.Spec.LabelSelector; selector != nil {
+			labelSelector = metav1.FormatLabelSelector(selector)
+		}
+
+		// Listing items for labelSelector (singular)
+		if len(orLabelSelectors) == 0 {
```

Comment:
it seems when `len(orLabelSelectors) == 0` and you choose to use `labelSelector`, the logic is almost identical to the case when there's only one element in the `orLabelSelectors`?

Why not combine them into one func, that takes the slice of label selectors as a param and returns the `unstructuredItems `

Reply:
Thanks @reasonerjt I have updated the PR, please take another look.